include/numerics/sparse_matrix.h

```diff
@@ -415,6 +415,23 @@ class SparseMatrix : public ReferenceCountedObject<SparseMatrix<T>>,
                          false); // false means DO NOT REUSE submatrix
   }
 
+  /**
+   * Similar to the above function, this function creates a \p
+   * submatrix which is defined by the indices given in the \p rows
+   * and \p cols vectors.
+   * Note: Both \p rows and \p cols can be unsorted;
```

Comment:
Would you mind updating the docs for the original `create_submatrix()` function to note that the rows and columns must be sorted in order for the routine to work correctly? There's apparenty no indication of this requirement at the moment. And I guess it means sorted both by rows and by column indices within each row? It would be great to clarify that as well...

Reply:
This is a good point. I will make the change to the docstring to avoid potential confusions in the future. Yes, both row and column indices need to be sorted in order for it to work properly.